py/dml/c_backend.py

```diff
@@ -349,6 +351,8 @@ def generate_hfile(device, headers, filename):
 
     print_device_substruct(device)
 
+    out(f'STATIC_ASSERT(sizeof(struct {crep.cname(device)}) <= UINT32_MAX);\n')
```

Comment:
We could also `out("// allows us to store device struct offsets in uint32, which saves space\n")` both for ourselves and for the perpetrator.